tests/unit/async_db/test_cursor.py

```diff
@@ -203,8 +255,14 @@ async def test_cursor_execute_error(
 ):
     """Cursor handles all types of errors properly."""
     for query in (
-        lambda: cursor.execute("select *"),
-        lambda: cursor.executemany("select *", []),
+        (
+            lambda: cursor.execute("select * from t"),
+            "server-side synchronous execute()",
+        ),
+        (
+            lambda: cursor.executemany("select * from t", []),
+            "server-side synchronous executemany()",
```

Comment:
We could probably add async counterparts in here as well, since errors should propagate the same way for them.

Reply:
I added them for rowcount and description. Did I miss anything?

Reply:
I mean that the same cases that are checked in here in `test_cursor_execute_error` can also be applied to the async queries. You can just add those to the list of cases in line258-264 and the same tests would be executed.

Reply:
Actually, I don't think I can add them yet, because `execute(query, async_exectution=True)` just returns a query id number. I can't add async tests until I have the ability to check the query status.

In the meantime what I'll do is add the failure modes below: `OperationalError` if `query_id` is not returned, and two `AsyncExecutionUnavailableError`s, one for when `Set use_standard_sql=1`, and as requested below, one when `len(queries) > 0 and async_execution`. The latter means I'll add another failure mode in `cursor.py`, as well.

Reply:
Ah I see, fair point!